packages/multi-test/src/app.rs

```diff
@@ -758,11 +780,17 @@ mod test {
             amount: coins(25, "eth"),
         };
         app.router
-            .execute(&mut cache, &app.block, owner.clone(), msg.into())
+            .execute(
+                app.api.as_ref(),
+                &mut cache,
+                &app.block,
+                owner.clone(),
+                msg.into(),
+            )
             .unwrap();
 
         // shows up in cache
-        let cached_rcpt = query_router(&app.router, &cache, &rcpt);
+        let cached_rcpt = query_router(&app.router, app.api.as_ref(), &cache, &rcpt);
```

Comment:
```suggestion
        let cached_rcpt = query_router(&app.router, &*app.api, &cache, &rcpt);
```
here and elsewhere?

Reply:
Honestly I personally find `as_ref()` more explicit, and what is more important - wider used also in this project. That is why I would keep it here for consistency.

Reply:
It's OK. It just seems nice to me to have all parameters referenced in more or less the same way (with `&`). But it's the same.

Reply:
Then I will change it, I somehow agree with you. Today I was working on other think at afternoon, but I will do it as first thing at the morning an merge this (as you approved I assume there are no other issues).

Reply:
Either way. I'll defer to both your judgement for handling boxes (feel free to update other usages if desired... maybe as a separate PR)